### PR TITLE
Save data in file 

### PR DESCRIPTION
**Pull request by HtetWaiYan7191**
### This pull request includes the following changes made by @HtetWaiYan7191  @porag-m06  @SAZZAD1Q2 

- Create JSON file to save data
- Create save method to save data in the file
- Create load_data method to show user when open the app
- Update README.md file

Thanks for reviewing our codes 😁

### CODE REVIEW

File: class/main.rb [hunk 1]
```diff
@@ -1,19 +1,57 @@
...
-    puts '7 - Exit'
+    puts '7 - Save data And Exit'
     user_choice = gets.chomp.to_i
     puts user_choice
     Execute.new(user_choice).execute
   end
+
+  # rubocop:disable Metrics/MethodLength
+  def load_data
+    books = JSON.parse(File.read('books.json'))
```
Comment (DammyShittu) [inline class/main.rb]: - [ ] There is an issue with your app. When starting it, it fails to load the books JSON file (though it exists) and crashes. Kindly fix this issue. Note that you should handle situations where the JSON files exist or not.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @HtetWaiYan7191, @SAZZAD1Q2 and @porag-m06,

## Highlights
✔️ Json files

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: class/main.rb (hunk 1 above)
Comment (porag-m06) [reply class/main.rb]: @DammyShittu  
# Hello, please check the screenshot below: 
image

Comment (mckent05) [reply class/main.rb]: - Good job done so far, but I am also experiencing the same issues as the previous reviewer, a reason for this is that you are loading your resources i.e books, person, and rentals from the wrong path. Please note that your `.json` files are in the `class` directory. Please fix this 🙏 
**OPTIONAL** - Also, when I  delete all `.json` and load the resources afresh, I get the error below. I suggest you include a condition in your code to initialize an empty array when there are no `.json` files. Please fix this 🙏 
rgqerg
**OPTIONAL** - Your load data method loads all resources when the app is initialized, this is not the expected behavior for the application. To enhance user experience, I suggest you only load a resource according to user input/options. Please refactor your load data method to ensure each resource has its own method when they are called. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi team, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
almost there


 ## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- You did well preserving your data in .json files 👏 👏 


 ## Required Changes ♻️
- You have done great so far, but please take your time to ensure that your project works accordingly, to the project requirements. Please fix this 👍 👍 
_Check the comments under the review._

## Optional suggestions

Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your code.


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by topeogunleye — APPROVED** [review]
### Hi @HtetWaiYan7191, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
- Fixed issue with loading the books JSON file. ✔️ 
 
## [Optional] suggestions:
👉 One optional suggestion for improvement is to follow the convention for naming your branch. Usually, branch names should use the past tense verb to indicate that the feature has been implemented. For example, instead of `preserve-data`, you could use `preserved-data`. This would make your branch name more consistent and clear for future pull requests. 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**